File: ui-preview/src/main/java/l/files/ui/preview/SizedColorDrawable.java
package l.files.ui.preview;

import android.graphics.drawable.ColorDrawable;

/**
 * A color drawable with fixed width and height.
 */
public final class SizedColorDrawable extends ColorDrawable {

    private int width;
    private int height;

    public SizedColorDrawable(int color, int width, int height) {
        super(color);
        this.width = width;
        this.height = height;
    }

    @Override
    public int getIntrinsicHeight() {
        return height;
    }

    @Override
    public int getIntrinsicWidth() {
        return width;
    }
}

File: ui-preview/src/main/java/l/files/ui/preview/SizedColorDrawable.kt
package l.files.ui.preview

import android.graphics.drawable.ColorDrawable

/**
 * A color drawable with fixed width and height.
 */
class SizedColorDrawable(
  color: Int,
  private val width: Int,
  private val height: Int
) : ColorDrawable(color) {

  override fun getIntrinsicHeight(): Int = height
  override fun getIntrinsicWidth(): Int = width

}
